### PR TITLE
refactor(*): deal with DB error in storage

### PR DESCRIPTION
**Pull request by sandyzhou**
The process panics if DB returns error except errors.ErrNotFound. If DB returns errors.ErrNotFound when it tries to get sth., returns nil data instead.

### CODE REVIEW

File: internal/ledger/blockchain.go [hunk 1]
```diff
@@ -6,6 +6,8 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/syndtr/goleveldb/leveldb/errors"
+
 	"github.com/meshplus/bitxhub/pkg/storage"
```
Comment (Alexader) [inline internal/ledger/blockchain.go:11]: import order?

File: internal/ledger/blockchain.go [hunk 2]
```diff
@@ -75,13 +79,12 @@ func (l *ChainLedger) GetBlockByHash(hash types.Hash) (*pb.Block, error) {
...
-	if err != nil {
-		return nil, err
+	v := l.blockchainStore.Get(compositeKey(transactionKey, hash.Hex()))
+	if v == nil {
+		return nil, errors.ErrNotFound
 	}
 	tx := &pb.Transaction{}
-	err = proto.Unmarshal(v, tx)
-	if err != nil {
+	if err := proto.Unmarshal(v, tx); err != nil {
```
Comment (4ever9) [inline internal/ledger/blockchain.go:87]: `tx.Unmarshal(v)`?

File: internal/ledger/ledger.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ var _ Ledger = (*ChainLedger)(nil)
 
 var (
 	ErrorRollbackTohigherNumber = fmt.Errorf("rollback to higher blockchain height")
```
Comment (4ever9) [inline internal/ledger/ledger.go:18]: `ErrorRollbackToHigherNumber `?

File: internal/ledger/blockchain.go (hunk 1 above)
Comment (sandyzhou) [reply internal/ledger/blockchain.go]: fixed

File: internal/ledger/ledger.go (hunk 3 above)
Comment (sandyzhou) [reply internal/ledger/ledger.go]: fixed

**Pull request merged**